co-simulation/fed/mosaic-infrastructure/src/main/java/org/eclipse/mosaic/fed/infrastructure/ambassador/InfrastructureInstanceManager.java

```diff
@@ -135,24 +151,99 @@ public V2xMessageTransmission onV2XMessageTx(InetAddress sourceAddr, CarmaV2xMes
     }
 
     /**
-     * Callback to be invoked when CARMA Platform receives a V2X Message from the NS-3 simulation
-     * @param rxMsg The V2X Message received
+     * Callback to be invoked when an RSU receives a V2X Message from the NS-3
+     * simulation
+     * 
+     * @param rxMsg   The V2X Message received
      * @param rxRsuId The Host ID of the vehicle receiving the data
-     * @throws RuntimeException If the socket used to communicate with the platform experiences failure
+     * @throws RuntimeException If the socket used to communicate with the platform
+     *                          experiences failure
      */
     public void onV2XMessageRx(byte[] rxMsg, String rxRsuId) {
-        if (!managedInstances.containsKey(rxRsuId))  {
+        if (!managedInstances.containsKey(rxRsuId)) {
             return;
         }
 
         InfrastructureInstance rsu = managedInstances.get(rxRsuId);
         try {
-            rsu.sendMsgs(rxMsg);
+            rsu.sendV2xMsg(rxMsg);
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
 
+    /**
+     * Callback to be invoked when a infrastructure instance receives a simulated
+     * object detection from a registered simulated sensors. Only send object detection
+     * to the infrastructure instance that contains the sensor that reported it.
+     * 
+     * @param detection Detected Object.
+     */
+    public void onDetectedObject(DetectedObject detection) {
+        for (InfrastructureInstance instance : managedInstances.values()) {
+            if (instance.containsSensor(detection.getSensorId())) {
```

Comment:
Do we have to identify the infrastructure instance registered with this particular sensor ID every time we are trying to forward detected object? Can we map the Detector to the sensor ID, so that we can directly access map with the sensor ID key. I think it might be faster to use Java standard library instead of implementing our own lookup with two nested loop if this lookup is frequently used as we constantly sending detected object?

Reply:
We can create a duplicate MAP for sensors to infrastructure IDs but I think this would only really matter if we have more that say 100 infrastructure instances. With less than 10 infrastructure instances I do not think this is a significant issue. Creating an additional map requires some small changes to how we instantiate each infrastructure instance and I think currently with less than 5 instances planned for concurrent deployment, I do not think it is a large issue. I think we can add a TODO statement to resolve this for larger deployments

Reply:
Agree with dan here, but also agree that it's not a big deal for small numbers. We're on order of O(n^2), but for small n that's still fine. I like your suggestion of adding a comment, a TODO in the doc comment would be great so that callers know the performance limitations.